src/basic/ordered-set.h

```diff
@@ -75,6 +75,18 @@ void ordered_set_print(FILE *f, const char *field, OrderedSet *s);
 #define ORDERED_SET_FOREACH(e, s) \
         _ORDERED_SET_FOREACH(e, s, UNIQ_T(i, UNIQ))
 
+#define ordered_set_clear_with_destructor(_s, _f)               \
+        ({                                                      \
+                void *_item;                                    \
+                while ((_item = ordered_set_steal_first(_s)))   \
+                        _f(_item);                              \
+        })
```

Comment:
hmm, i think this should probably evaluate to _s in the end, so that you can write:

ordered_set_put(ordered_set_clear_with_destructor(s, f), x);

or so?

i.e. add a line `_f;` to the end of the expression